Tests exist but aren’t run automatically on GitHub

## Description'
The repo already has automated tests (for example graph matching and UI safety checks), but there’s no GitHub Actions workflow that runs them when someone opens a pull request. Without CI, contributors and reviewers can’t quickly see if a change broke graph logic or frontend.

## Changes Expected

Add a workflow that installs dependencies and runs `pytest` on pushes and pull requests. If some installs are heavy (for example optional NLP models), the issue description can note documenting skips or a lighter “CI requirements” path—whatever keeps the first version mergeable and maintainable.

## Test Failure Found While Setting Up CI

While validating the new GitHub Actions test workflow, the test suite exposed an existing failure in the graph traversal/scoring logic.

### Failing Test
`tests/test_graph_matching.py::test_multiple_symptoms_all_matched`

### Failure Summary
The test expects multiple symptoms (`fever`, `sore throat`, and `body aches`) to all contribute to the final condition score. However, the top result returns a `raw_score` of `0.472`, which suggests that only a single symptom contribution is being counted, or that the score is not being accumulated as the test expects.

### Error
```text
AssertionError: raw_score=0.472 — only one symptom seems to have contributed
assert 0.472 > 1.0
```

<img width="1911" height="650" alt="Image" src="https://github.com/user-attachments/assets/479967b0-4653-4b95-a893-057258752011" />

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -1,0 +1,32 @@
+name: Run Tests
+
+on:
+    push:
+        branches:
+            - main
+    pull_request:
+
+permissions:
+    contents: read
+
+jobs:
+    pytest:
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v4
+            
+            - name: Set up Python
+              uses: actions/setup-python@v5
+              with:
+                python-version: "3.12"
+                cache: "pip"
+            
+            - name: Install lightweight CI dependencies
+              run: |
+                python -m pip install --upgrade pip
+                pip install pytest networkx
+
+            - name: Run tests
+              run: python -m pytest -q tests

